cmake/kokkos_tribits.cmake

```diff
@@ -352,6 +352,7 @@ FUNCTION(KOKKOS_SET_LIBRARY_PROPERTIES LIBRARY_NAME)
     ""
     ${ARGN})
 
+  IF(NOT Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE)
   IF(${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18")
```

Comment:
```suggestion
  IF(NOT Kokkos_ENABLE_COMPILE_AS_CMAKE_LANGUAGE AND ${CMAKE_VERSION} VERSION_GREATER_EQUAL "3.18")
```

Reply:
note the else clause is only for CMAKE 3.18 not for CMAKE_LANGUAGE